Userland/Services/SystemServer/Service.cpp

```diff
@@ -273,7 +310,7 @@ void Service::did_exit(int exit_code)
     s_service_map.remove(m_pid);
     m_pid = -1;
 
-    if (!m_keep_alive)
+    if (!m_keep_alive || m_force_stopped)
```

Comment:
This should probably check whether the service is enabled rather than whether it has been explicitly stopped?